Signal/src/ViewControllers/ConversationView/MessagesViewController.m

```diff
@@ -109,6 +110,40 @@
     kMediaTypeVideo,
 } kMediaTypes;
 
+@protocol OWSMessagesCollectionViewFlowLayoutDelegate <NSObject>
+
+- (BOOL)isSpecialItemAtIndexPath:(NSIndexPath *)indexPath;
+
+@end
+
+#pragma mark -
+
+@interface OWSMessagesCollectionViewFlowLayout : JSQMessagesCollectionViewFlowLayout
+
+@property (nonatomic, weak) id<OWSMessagesCollectionViewFlowLayoutDelegate> delegate;
+
+@property (nonatomic) CGRect lastBounds;
+
+@end
+
+#pragma mark -
+
+@implementation OWSMessagesCollectionViewFlowLayout
+
+- (CGSize)sizeForItemAtIndexPath:(NSIndexPath *)indexPath
+{
+    // The unread indicator should be sized according to its desired size.
+    if ([self.delegate isSpecialItemAtIndexPath:indexPath]) {
```

Comment:
In subsequent PRs, we use this same technique to do custom sizing of the system messages.